video/out/vo_x11.c

```diff
@@ -151,8 +153,16 @@ static void freeMyXImage(struct priv *p, int foo)
         XDestroyImage(p->myximage[foo]);
         shmdt(p->Shminfo[foo].shmaddr);
     } else {
-        if (p->myximage[foo])
+        if (p->myximage[foo]) {
+            if (p->myximage[foo]->data) {
+                // XDestroyImage frees both the image and the data. but the
+                // data was calloc()-ed by us. so free it ourselves and set the
+                // pointer to NULL to avoid double free.
```

Comment:
consider:
```suggestion
                // XDestroyImage would free the data too, but do it ourselves for portability reasons.
```

Reply:
It's not exactly "portability" reasons but to avoid depending on Xlib internals. How about this:

```
// Since XFree just calls free, XDestroyImage would free the data too even
// though it was calloc-ed. But avoid depending on Xlib's internal behavior.
```

Reply:
isn't that the definition of portability? we know Xlib will free the data, but we don't want to rely on _how_ it does that

Reply:
I was thinking that since (AFAIK) XFree has always called `free`, it wouldn't be "non-portable" in practice. But sure, that works too, I don't feel too strongly about it.

Also noticed the null check was redundant since `free(NULL)` is no-op, so removed that as well.